elements/lisk-p2p/src/inbound_peer.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+
+import {
+	Peer,
+	PeerConfig,
+	REMOTE_EVENT_MESSAGE,
+	REMOTE_EVENT_RPC_REQUEST,
+	SCServerSocketUpdated,
+} from './peer';
+
+import { P2PDiscoveredPeerInfo } from './p2p_types';
+
+import { SCServerSocket } from 'socketcluster-server';
+
+export class InboundPeer extends Peer {
+	protected _socket: SCServerSocketUpdated | undefined;
+
+	public constructor(
+		peerInfo: P2PDiscoveredPeerInfo,
+		peerConfig?: PeerConfig,
+		peerSocket?: SCServerSocket,
+	) {
+		super(peerInfo, peerConfig);
+		this._socket = peerSocket ? peerSocket : undefined;
+		if (this._socket) {
+			this._bindHandlersToInboundSocket(this._socket);
+		}
+	}
+
+	public set inboundSocket(scServerSocket: SCServerSocket) {
```

Comment:
Mayeb this setter should be renamed to just `socket` instead of `inboundSocket`?